src/nifgen/metadata/functions_addon.py

```diff
@@ -84,8 +84,17 @@
     'ConfigureDigitalLevelScriptTrigger':       { 'parameters': { 3: { 'enum': 'TriggerWhen',               }, }, },
 }
 
+# Need to manually set some repeated capabilities
+functions_rep_caps = {
+    'ConfigureDigitalEdgeScriptTrigger':        { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
+    'ConfigureDigitalLevelScriptTrigger':       { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
+    'ConfigureSoftwareEdgeScriptTrigger':       { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
+    'DisableScriptTrigger':                     { 'parameters': { 1: { 'is_repeated_capability': True, }, }, },
+    'SendSoftwareEdgeTrigger':                  { 'parameters': { 2: { 'is_repeated_capability': True, }, }, },
```

Comment:
Should `'SendSoftwareEdgeTrigger'` change this? Is it used differently?

Reply:
This is the C signature:

``` C
ViStatus niFgen_SendSoftwareEdgeTrigger (ViSession vi, viInt32 trigger, viConstString triggerID);
```

Looking at [documentation](http://zone.ni.com/reference/en-XX/help/370524T-01/fgencref/cvinifgen_sendsoftwareedgetrigger/), the `trigger` parameter should be an enum and the `triggerID` parameter should allow for string `"None"` as well. Unless the driver treats `""` and `VI_NULL` to mean the same as `"None"`.

Reply:
There is a `Trigger` enum in enums_addon. Should I use that? It was waiting on #624 which is closed.

I don't think repeated capabilities as currently working will support "None". It will become "ScriptTriggerNone" which is not what we want for this particular use case.

Should maybe this function not use the repeated capability? It seems like a different use.

Reply:
> Should maybe this function not use the repeated capability? It seems like a different use.

I definitely don't think it works as it is in the PR right now:

``` python
send_software_edge_trigger(trigger)
```

Valid use cases would be:

``` python
session.script_triggers[1].send_software_edge_trigger(nifgen.Trigger.SCRIPT)
session["None"].send_software_edge_trigger(nifgen.Trigger.START)
```

If you use the `script_triggers container` then trigger **must be** `nifgen.Trigger.SCRIPT `. That's silly and redundant.

User has no good way of knowing `session.script_triggers["None"]` is what myst be passed if you want to send a start software trigger.

If we didn't make `viConstString triggerID` a repeated capability:

``` python
session.send_software_edge_trigger(nifgen.Trigger.SCRIPT, "ScriptTrigger1")
session.send_software_edge_trigger(nifgen.Trigger.START, "None")
```

User would need to know the magic strings. This is weird but we can live with it.

Ideally, arguably, usage should be:

``` python
session.script_triggers[1].send_software_edge_trigger()
session.send_software_edge_trigger() # sends start trigger
```

No enum needed. I'm not sure this is the best way to go, and I think it would require at the very least a fancy wrapper.

Do other APIs have the same signature for sending software triggers?

Reply:
For expediency, I suggest removing the repeated capability from this function and opening an issue to track it / decide on it separately.